src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -144,16 +169,14 @@ private static JsonNode validateAndParseSemiStructuredAsJsonTree(
    * @return JSON string representing the input
    */
   static String validateAndParseVariant(String columnName, Object input, long insertRowIndex) {
-    JsonNode node =
-        validateAndParseSemiStructuredAsJsonTree(columnName, input, "VARIANT", insertRowIndex);
+    final String result =
+        validateAndParseSemiStructured(columnName, input, "VARIANT", insertRowIndex);
 
-    // Missing nodes are not valid json, ingest them as NULL instead
-    if (node.isMissingNode()) {
+    // Empty json strings are ingested as nulls
```

Comment:
Does a "" get replaced with NULL? If so that's us corrupting data, they're not really equivalent values.

Reply:
They are equivalent. `select parse_json('');` yields `null`. `''` and `'""'` are different variants.

Reply:
To ingest an empty string, customers need to ingest Java string `"\"\""`.